### PR TITLE
Delete item

### PR DESCRIPTION
**Pull request by Matt-Kou**

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -207,6 +207,33 @@ def get_item(order_id, item_id):
...
+    """
+    app.logger.info("Request to delete an Item %s from Order with id: %s", item_id, order_id)
+    
+    # See if the order exists and abort if it doesn't
+    order = Order.find(order_id)
+    if not order:
+        abort(
+            status.HTTP_404_NOT_FOUND,
+            f"Order with id '{order_id}' could not be found.",
+        )
```
Comment (MahamArif) [inline service/routes.py:226]: As discussed with professor, there is no need to check whether the order exists in delete case. The API will return HTTP_204_NO_CONTENT in all cases.

File: tests/test_routes.py [hunk 2]
```diff
@@ -349,4 +383,4 @@ def test_get_item_order_not_found(self):
         It should not Read an item when the order is not found
         """
         resp = self.app.get(f"{BASE_URL}/0/items/0")
-        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
```
Comment (MahamArif) [inline tests/test_routes.py:386]: Please don't remove new line from the end of file. Run `make lint` command to make sure that your code doesn't introduce any linting issues.

File: service/routes.py [hunk 3]
```diff
@@ -207,6 +207,28 @@ def get_item(order_id, item_id):
...
+@app.route("/orders/<int:order_id>/items/<int:item_id>", methods=["DELETE"])
+def delete_item(order_id, item_id):
+    """
+    Delete an item from an order
+    """
+    app.logger.info("Request to delete an Item %s from Order with id: %s", item_id, order_id)
+    
+    # See if the order exists and abort if it doesn't
+    order = Order.find(order_id)
+    if order:
```
Comment (MahamArif) [inline service/routes.py:222]: You can directly check if item exists in db. No need to add an extra check for order

Comment (MahamArif) [inline service/routes.py:222]: You can directly check if item exists in db. No need to add an extra check for order.

**Review by MahamArif — CHANGES_REQUESTED** [review]

File: service/routes.py (hunk 3 above)
Comment (Matt-Kou) [reply service/routes.py:222]: I checked the order ID becaues I think if the order ID is wrong, I shouldn't delete the item?

**Pull request merged**